contributed_definitions/nyaml/NXpeak_model.yaml

```diff
@@ -0,0 +1,140 @@
+category: base
+doc: |
+  Peak fitting of a one-dimensional spectrum.
+symbols:
+  doc: |
+    The symbols used in the schema to specify e.g. dimensions of arrays.
+  n_peaks: |
+    Number of peaks.
+type: group
+NXpeak_model(NXprocess):
+  region_REGION(NXfit_region):
+    doc: |
+      Subset of the abscisssa that is used for fitting.
+      It is envisioned that regions are labeled as regions_0, regions_1, and so on.
+  peak_PEAK(NXpeak):
+    doc: |
+      One spectral peak of the peak model.
+      It is envisioned that peaks are labeled as peak_0, peak_1, and so on.
+  background(NX_background):
+    doc: |
+      Description of the function used for background subtraction.
+  envelope(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Resulting envelope of all NXpeak instances.
+      
+      Should have the same dimension as the `position` axes of the individual peaks.
+  figure_of_merit(NX_NUMBER):
+    unit: NX_UNITLESS
+    doc: |
+      Figure-of-merit, e.g., :math:`\Chi^2`, or the standard deviation of the residuals,
+      to determine the goodness of fit, i.e., how well the peak model fits the measured
+      observations.
+      
+      This value (which is a single number) is oftenused to guide adjustments to the
+      fitting parameters in the peak fitting process.
```

Comment:
There is no place to store what the actual FOM corresponds to. Maybe add several explicit values (Chi2, red_CHi2, R2, etc.)

Reply:
Like so:
```yaml
figure_of_merit(NX_NUMBER):
  unit: NX_UNITLESS
  doc: |
    Figure-of-merit to determine the goodness of fit, i.e., how well the peak model
    fits the measured observations.
    
    This value (which is a single number) is oftenused to guide adjustments to the
    fitting parameters in the peak fitting process.
  \@metric:
    doc: |
      Metric used to determine the goodness of fit. Examples include:
        - :math:`\Chi^2`, the standard deviation of the residuals
        - reduced :math:`\Chi^2`:, :math:`\Chi^2`: per degree of freedom
        - :math:`R^2`, the coefficient of determination
```

Reply:
Fine, but then one cannot add multiple of these, no? (e.g. chi2 and R2)

Reply:
Or we go with figure_of_merit_METRIC? I just have a hard time coming up with all possible metrics if I want to put them there as fields explicitly.